qcodes/instrument_drivers/tektronix/TPS2012_channels.py

```diff
@@ -0,0 +1,344 @@
+
+        preamble = self._preambleparser(preamblestr)
+        # the raw curve data starts with a header containing the char #
+        # followed by on digit giving the number of digits in the len of the
+        # array in bytes
+        # and the length of the array. I.e. the string #45000 is 5000 bytes
+        # represented by 4 digits.
+        total_number_of_bytes = preamble['no_of_bytes']*preamble['no_of_points']
+        raw_data_offset = 2 + len(str(total_number_of_bytes))
+        curvestr = curvestr[raw_data_offset:-1]
+        rawcurve = self._binaryparser(curvestr)
+
+        yoff = preamble['y_offset']
+        yoff -= 2**15  # data width 2
+        ymult = preamble['y_multiplier']
+        ydata = ymult*(rawcurve-yoff)
+        assert len(ydata) == preamble['no_of_points']
+        xstart = preamble['x_zero']
+        xinc = preamble['x_incr']
+        xdata = np.linspace(xstart, len(ydata)*xinc+xstart, len(ydata))
+        return xdata, ydata, preamble['no_of_points']
+
+class TPS2012Channel(InstrumentChannel):
+
+    def __init__(self, parent, name, channel):
+        super().__init__(parent, name)
+
+        self.add_parameter('scale',
+                           label='Channel {} Scale'.format(channel),
+                           unit='V/div',
+                           get_cmd='CH{}:SCAle?'.format(channel),
+                           set_cmd='CH{}:SCAle {}'.format(channel, '{}'),
+                           get_parser=float
+                           )
+        self.add_parameter('position',
+                           label='Channel {} Position'.format(channel),
+                           unit='div',
+                           get_cmd='CH{}:POSition?'.format(channel),
+                           set_cmd='CH{}:POSition {}'.format(channel, '{}'),
+                           get_parser=float
+                           )
+        self.add_parameter('curvedata',
+                           channel=channel,
+                           parameter_class=ScopeArray,
+                           )
+        self.add_parameter('state',
+                           label='Channel {} display state'.format(channel),
+                           set_cmd='SELect:CH{} {}'.format(channel, '{}'),
+                           get_cmd=partial(self._get_state, channel),
+                           val_mapping={'ON': 1, 'OFF': 0},
+                           vals=vals.Enum('ON', 'OFF')
+                           )
+
+    def _get_state(self, ch):
+        """
+        get_cmd for the chX_state parameter
+        """
+        # 'SELect?' returns a ';'-separated string of 0s and 1s
+        # denoting state display state of ch1, ch2, ?, ?, ?
+        # (maybe ch1, ch2, math, ref1, ref2 ..?)
+        selected = list(map(int, self.ask('SELect?').split(';')))
+        state = selected[ch - 1]
+        return state
+
+class TPS2012WithChannels(VisaInstrument):
+    """
+    This is the QCoDeS driver for the Tektronix 2012B oscilloscope.
```

Comment:
The class name and docstring should agree; either this is THE QCoDeS driver for TPS2012 or it's the channelised version.